services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_api.py

```diff
@@ -452,13 +454,18 @@ def _count_containers(item: dict[str, Any]) -> int:
     return {x["Name"]: _count_containers(x) for x in filtered_networks}
 
 
-async def try_to_remove_network(network_name: str) -> None:
+async def try_to_remove_network(network_name: str) -> bool:
     async with docker_client() as client:
         network = await client.networks.get(network_name)
+
+        # if a project network for the current project has no more
+        # containers attached to it (because the last service which
+        # was using it was removed), also removed the network
         try:
             return await network.delete()
         except aiodocker.exceptions.DockerError:
             log.warning("Could not remove network %s", network_name)
```

Comment:
IMO same here. If

```python
if not await try_to_remove_network(network_name)
    log.warning("Could not remove network %s", network_name)
```

or

```python
try:
   await try_to_remove_network(network_name)
except NetworkError as e:
    log.warning("Could not remove network %s: e", network_name, e)
```

Reply:
I do not agree with you here. The function is used like so:

```python
await logged_gather(
    *[
        try_to_remove_network(network_name)
        for network_name, container_count in used_projects_networks.items()
        if container_count == 0
    ]
)
```

I think it makes it will only add code and not help with readability.

Reply:
I see. Yes, it makes sense